spec/views/user_show_spec.rb

```diff
@@ -0,0 +1,50 @@
+require 'rails_helper'
+
+RSpec.describe 'User show page - ', type: :feature do
+  before :each do
+    @user1 = User.create(
+      name: 'User 1',
+      photo: 'https://unsplash.com/photos/Hekapoo',
+      bio: 'User1 bio',
+      posts_counter: 0
+    )
+    @post1 = Post.create(author: @user1, title: 'post 1', text: 'new post 1', comments_counter: 0, likes_counter: 0)
+    @post2 = Post.create(author: @user1, title: 'post 2', text: 'new post 2', comments_counter: 0, likes_counter: 0)
+    @post3 = Post.create(author: @user1, title: 'post 3', text: 'new post 3', comments_counter: 0, likes_counter: 0)
+    @post4 = Post.create(author: @user1, title: 'post 4', text: 'new post 4', comments_counter: 0, likes_counter: 0)
+    @post5 = Post.create(author: @user1, title: 'post 5', text: 'new post 5', comments_counter: 0, likes_counter: 0)
+    @comment1 = Comment.create(author: @user1, post: @post1, text: 'my comment')
+    visit user_path(@user1)
+  end
+
+  it 'I can see the user\'s profile picture' do
+    expect(page).to have_css("img[src='#{@user1.photo}']")
+  end
+
+  it 'I can see the user\'s username.' do
+    expect(page).to have_content(@user1.name)
+  end
+
+  it 'I can see the user\'s last 3 posts.' do
+    expect(page).to have_content(@post3.text)
+    expect(page).to have_content(@post4.text)
+    expect(page).to have_content(@post5.text)
+  end
+
+  it 'I can see the number of posts the user has written.' do
+    expect(page).to have_content("Number of posts: #{@user1.posts_counter}")
+  end
+
+  it 'I can see the user\'s bio.' do
+    expect(page).to have_content(@user1.bio)
+  end
+
+  it 'I can see a button that lets me view all of a user\'s posts.' do
+    expect(page).to have_content('See all posts')
+  end
+
+  it "When I click to see all posts, it redirects me to the user's post's index page." do
+    click_link 'See all posts'
+    expect(page).to have_current_path(user_posts_path(@user1))
+  end
+end
```

Comment:
- [ ] Kindly add test for this assertion:

  - Tests that redirects to the post's show page when you click on a user's post.

Reply:
@julie-ify Thank you for your comments. I'd like to ask you if this test is not a part of the post_index view. In this case I created this test in the post_index_spec.rb file like 'When I click on a post, it redirects me to that posts show page.'